ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsIDPConfig.scala

```diff
@@ -0,0 +1,165 @@
+  }
+
+  it should "check if identity provider config's issuer exists" in {
+    val cfg = config()
+    executeSql(tested.idpConfigByIssuerExists(cfg.issuer)) shouldBe false
+    executeSql(tested.createIdentityProviderConfig(cfg))
+    executeSql(tested.idpConfigByIssuerExists(cfg.issuer)) shouldBe true
+  }
+
+  it should "check if identity provider config by id exists" in {
+    val cfg = config()
+    executeSql(tested.idpConfigByIdExists(cfg.identityProviderId)) shouldBe false
+    executeSql(tested.createIdentityProviderConfig(cfg))
+    executeSql(tested.idpConfigByIdExists(cfg.identityProviderId)) shouldBe true
+  }
+
+  it should "fail to update issuer for non existing identity provider config" in {
+    executeSql(tested.updateIssuer(randomId(), "whatever")) shouldBe false
+    executeSql(tested.updateIssuer(randomId(), "")) shouldBe false
+  }
+
+  it should "fail to update isDeactivated for non existing identity provider config" in {
+    executeSql(tested.updateIsDeactivated(randomId(), true)) shouldBe false
+    executeSql(tested.updateIsDeactivated(randomId(), false)) shouldBe false
+  }
+
+  it should "fail to update jwksURL for non existing identity provider config" in {
+    executeSql(
+      tested.updateJwksURL(randomId(), JwksUrl("http://example.com/jwks.json"))
+    ) shouldBe false
+    executeSql(
+      tested.updateJwksURL(randomId(), JwksUrl("http://example2.com/jwks.json"))
+    ) shouldBe false
+  }
+
+  it should "fail to update identity provider config issuer attribute to non-unique issuer" in {
+    val cfg1 = config()
+    val cfg2 = config()
+    executeSql(tested.createIdentityProviderConfig(cfg1))
+    executeSql(tested.createIdentityProviderConfig(cfg2))
+    assertThrows[SQLException] {
+      executeSql(tested.updateIssuer(cfg1.identityProviderId, cfg2.issuer))
+    }
+  }
+
+  it should "fail to create identity provider config with non-unique issuer" in {
+    val cfg1 = config()
+    val cfg2 = config()
+    executeSql(tested.createIdentityProviderConfig(cfg1))
+    assertThrows[SQLException] {
+      executeSql(tested.createIdentityProviderConfig(cfg2.copy(issuer = cfg1.issuer)))
+    }
+  }
+
+  it should "fail to create identity provider config with non-unique id" in {
+    val cfg1 = config()
+    val cfg2 = config()
+    executeSql(tested.createIdentityProviderConfig(cfg1))
+    assertThrows[SQLException] {
+      executeSql(
+        tested.createIdentityProviderConfig(cfg2.copy(identityProviderId = cfg1.identityProviderId))
+      )
+    }
+  }
+
+  it should "get all identity provider configs ordered by id" in {
+    val cfg1 = config()
+    val cfg2 = config()
+    val cfg3 = config()
+    executeSql(tested.createIdentityProviderConfig(cfg1))
+    executeSql(tested.createIdentityProviderConfig(cfg2))
+    executeSql(tested.createIdentityProviderConfig(cfg3))
+
+    executeSql(
+      tested.listIdentityProviderConfigs()
+    ) should contain theSameElementsAs Vector(cfg1, cfg2, cfg3)
```

Comment:
afaict you're not checking the order of elements why the test case's description claims order by id